ledger/sandbox/src/main/scala/com/digitalasset/platform/index/StandaloneIndexerServer.scala

```diff
@@ -10,54 +10,65 @@ import com.digitalasset.platform.common.logging.NamedLoggerFactory
 import com.digitalasset.platform.common.util.{DirectExecutionContext => DEC}
 import com.digitalasset.platform.index.config.{Config, StartupMode}
 
-import scala.concurrent.{Await, Future, Promise}
 import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future, Promise}
+import scala.util.control.NonFatal
 
 // Main entry point to start an indexer server.
 // See v2.ReferenceServer for the usage
 object StandaloneIndexerServer {
+  private val asyncTolerance: FiniteDuration = 10.seconds
+
+  private implicit val executionContext: ExecutionContext = DEC
 
   def apply(
       readService: ReadService,
       config: Config,
       loggerFactory: NamedLoggerFactory,
-      metrics: MetricRegistry): Future[AutoCloseable] = {
-
+      metrics: MetricRegistry,
+  ): Future[AutoCloseable] = {
     // ActorSystem name not allowed to contain daml-lf LedgerString characters ".:#/ "
     val actorSystem = ActorSystem(
       "StandaloneIndexerServer-" + config.participantId.filterNot(".:#/ ".toSet))
-    val asyncTolerance: FiniteDuration = 10.seconds
-    val indexerFactory = JdbcIndexerFactory(metrics, loggerFactory)
-    val indexer =
-      RecoveringIndexer(
-        actorSystem.scheduler,
-        asyncTolerance,
-        indexerFactory.asyncTolerance,
-        loggerFactory)
 
-    val initializedIndexerFactory = config.startupMode match {
-      case StartupMode.MigrateOnly | StartupMode.MigrateAndStart =>
-        indexerFactory.migrateSchema(config.jdbcUrl)
-      case StartupMode.ValidateAndStart => indexerFactory.validateSchema(config.jdbcUrl)
-    }
+    val indexerFactory = JdbcIndexerFactory(metrics, loggerFactory)
+    val indexer = RecoveringIndexer(
+      actorSystem.scheduler,
+      asyncTolerance,
+      indexerFactory.asyncTolerance,
+      loggerFactory)
 
     val promise = Promise[Unit]
 
-    config.startupMode match {
-      case StartupMode.MigrateOnly => promise.success(())
-      case StartupMode.MigrateAndStart | StartupMode.ValidateAndStart =>
-        indexer.start { () =>
-          {
-            val createF = initializedIndexerFactory
+    def startIndexer(initializedIndexerFactory: JdbcIndexerFactory[Initialized]): Future[Unit] = {
+      println("Starting the indexer.")
```

Comment:
This should probably be a log statement.

Reply:
Whoops, debugging line left in. I will delete it unless you think it's useful.

Reply:
Fixed.